### ISSUE TITLE
[BUG] SecureParameter on Vault with no secrets

### ISSUE THREAD

**Issue by jsacapdev** [Azure/PSRule.Rules.Azure#3347]
### Existing rule

Azure.Deployment.SecureParameter

### Description of the issue

``` bash
Using PSRule v2.9.0
Using PSRule.Rules.Azure v1.43.0
````

When attempting to deploy a key vault through AVM without any secrets, I get a FAIL around using secure parameters for any parameter that contain sensitive. The thing is that the deployment does not add any secrets, so I am a bit confused:

``` bicep
module vault 'br/public:avm/res/key-vault/vault:0.12.1' = {
  scope: az.resourceGroup(resourceGroupName)
  name: 'vault'
  params: {
    name: resourceNames.keyvault
    location: resourceGroup.outputs.location
    tags: tags
    enablePurgeProtection: true
    enableSoftDelete: true
    enableVaultForDeployment: true
    enableVaultForDiskEncryption: false
    enableVaultForTemplateDeployment: true
    enableRbacAuthorization: true
    publicNetworkAccess: 'Disabled'
    sku: 'standard'
    networkAcls: {
      defaultAction: 'Deny'
      bypass: 'AzureServices'
    }
  }
}
```

While I could add a supression, I dont believe I should need to. As there are no secrets being deployed.

### Error messages

``` bash
-> vault : Microsoft.Resources/deployments [7/8]

    [FAIL] Azure.Deployment.SecureParameter (AZR-000408)
    | Template: /workspaces/infra/configuration/bicep/main.bicep:0:0

    | RECOMMEND:
    | Consider using secure parameters for any parameter that contain sensitive
    | information.

    | REASON:
    | - Path secrets: The parameter 'secrets' with type 'array' is not secure.

    | HELP:
    | - https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.Deployment.SecureParameter/
```

### Reproduction

In a devcontainer, configure the modules and then run assert rule:

``` pwsh
Install-Module -Name 'PSRule.Rules.Azure' -Repository PSGallery -Scope CurrentUser -Force -AcceptLicense;
Install-Module -Name 'Az.Resources' -Repository PSGallery -Scope CurrentUser -Force -AcceptLicense;

# Run PSRule validation
Assert-PSRule -InputPath "/workspaces/infra/$Entity/" `
              -Option "/workspaces/infra/ps-rule.yaml" `
              -OutputPath $resultsPath `
              -OutputFormat NUnit3 `
              -Outcome Fail
```

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.43.0

**Comment by BernieWhite** [comment]
Hi @jsacapdev, thanks for reporting the issue.

The rule `Azure.Deployment.SecureParameter` specifically checks your module deployment instead of the key vault it itself, looking for potential cases of parameters that should be marked secure that aren't.

It's failing because the AVM module you are consuming `br/public:avm/res/key-vault/vault:0.12.1`, has a parameter called `secrets` and this secrets parameter is not a secure type it is an array type. This is still present in your deployment regardless of if you use it.

The tricky bit is that the `secrets` parameter is a custom type `secretType` which has a secure property marked further down for the actual secret value.

I agree we can improve the experience here to recognise that only part of the parameter could potentially contain sensitive information.

---

To work around this issue, you could configure the `AZURE_DEPLOYMENT_NONSENSITIVE_PARAMETER_NAMES` configuration value as specified in the docs: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.Deployment.SecureParameter/

Alternatively, you can suppress this rule by the options here: https://azure.github.io/PSRule.Rules.Azure/concepts/suppression/

I hope that helps explain the issue and give you a path forward.

**Comment by jsacapdev** [comment]
Thank you @BernieWhite. I have opted to the take the `AZURE_DEPLOYMENT_NONSENSITIVE_PARAMETER_NAMES`. I will leave this item open for you to evaluate if any further changes are required. If not or this is being tracked elsewhere feel free to close.
